internal/types/compare.go

```diff
@@ -255,3 +254,130 @@ func compareNumbers(a float64, b int64) CompareResult {
 
 	return CompareResult(bigA.Cmp(bigB))
 }
+
+// dataTypeOrderResult represents the comparison order of data types.
+type dataTypeOrderResult uint8
+
+const (
+	_ dataTypeOrderResult = iota
+	nullDataType
+	nanDataType
+	numbersDataType
+	stringDataType
+	objectDataType
+	arrayDataType
+	binDataType
+	objectIdDataType
+	booleanDataType
+	dateDataType
+	timestampDataType
+	regexDataType
+)
+
+// defineDataType define which type has value and returns a sequence the type has.
```

Comment:
```suggestion
// defineDataType returns a sequence for build-in type.
```

Reply:
would you be so kind to add a comment about Document and Array types?

Reply:
A comment for `detectType` has been changed.

A comment for Document and Array - do you mean like TODO?

Reply:
I just didn't understand - the func was accepting `any` and the comment specified that the func is for build-in type, but `Document` and `Array`, which are in build-in type, were not processed in the function.

Reply:
I'm going to engage them (`Document` and `Array` types) in another PR.